sigstore/_cli.py

```diff
@@ -650,16 +650,16 @@ def _sign(args: argparse.Namespace) -> None:
     elif args.fulcio_url == DEFAULT_FULCIO_URL and args.rekor_url == DEFAULT_REKOR_URL:
         signing_ctx = SigningContext.production()
     else:
-        # Assume "production" keys if none are given as arguments
-        updater = TrustUpdater.production()
+        # Assume "production" trust root if no keys are given as arguments
+        trusted_root = TrustedRoot.production()
         if args.ctfe_pem is not None:
             ctfe_keys = [args.ctfe_pem.read()]
         else:
-            ctfe_keys = updater.get_ctfe_keys()
+            ctfe_keys = trusted_root.get_ctfe_keys()
         if args.rekor_root_pubkey is not None:
             rekor_keys = [args.rekor_root_pubkey.read()]
         else:
-            rekor_keys = updater.get_rekor_keys()
+            rekor_keys = trusted_root.get_rekor_keys()
```

Comment:
there's a slight change of functionality here (in contrast to what I said in the PR message):
this now leads to TUF repo getting  refreshed even if all keys are provided via command line... I'll think about how to avoid that

Reply:
Making sure I understand: this causes the TUF repo to get refreshed because `TrustedRoot.production()` is doing the refresh, rather than deferring it to one of the `get_*_keys()` calls, right?

Reply:
Yes. In the future (if the local keys can only be provided via --trust-root) this would not be an issue but in this transition state it could be

Reply:
Makes sense, thanks. We can hold off if you feel strongly here, but I'm personally okay with this -- we intend to remove these flags anyways, so IMHO we can deal with a temporary unnecessary network request.

Reply:
Let's leave it like this then and move on with next steps